src/crewai/agent.py

```diff
@@ -608,6 +618,26 @@ def _render_text_description(self, tools: List[Any]) -> str:
 
         return description
 
+    def _inject_date_to_task(self, task):
+        """Inject the current date into the task description if inject_date is enabled."""
+        if self.inject_date:
+            from datetime import datetime
+            try:
+                valid_format_codes = ['%Y', '%m', '%d', '%H', '%M', '%S', '%B', '%b', '%A', '%a']
+                is_valid = any(code in self.date_format for code in valid_format_codes)
+                
+                if not is_valid:
+                    raise ValueError(f"Invalid date format: {self.date_format}")
+                
+                current_date: str = datetime.now().strftime(self.date_format)
+                task.description += f"\n\nCurrent Date: {current_date}"
```

Comment:
Would it be relevant if we also consider invalid if not enough format codes are provided to securely infer a date?

e.g. 
```python
date_aware_agent = Agent(
    ...
    inject_date=True,
    date_format="%M, %m, %S", # minute, month(number), second
    ...
)
```

Just out of curiosity, we probably don't want to bother about those cases too much, since it's more of a misuse. Maybe adding a warning?

Reply:
def a nice to have, will do a follow up